jwst/lib/basic_utils.py

```diff
@@ -2,6 +2,44 @@
 import re
 
 
+def deprecate_class(new_class,
+                    message='"{old_class}" is deprecated and will be removed. Use {new_class}'):
+    """Deprecate a class in favor of another class
+
+    Parameters
+    ----------
+    new_class : class
+        The class/object that should be used instead
+
+    message : str
+        The deprecation warning message
+    """
+
+    # Implement the inner decorator
+    def _decorator(old_class):
+
+        def init(self, *args, **kwargs):
+            import warnings
+            warnings.simplefilter('default')
+            warnings.warn(message.format(old_class=old_class.__name__, new_class=new_class.__name__),
+                          category=DeprecationWarning)
+            new_class.__init__(self, *args, **kwargs)
+
+        # Create the class
+        deprecated = type(
```

Comment:
Ah, here's the answer to my question above. A `RampModel` (new_class) is indeed returned. Never mind ...

Reply:
To be more specific, the class returned is called `MIRIRampModel`. However, as noted, it is a `RampModel` but with the warning tacked into the init. Essentially just subclassing, but programmatically.